app/pathogen/arbovirus/analyze/recharts/visualization-header.tsx

```diff
@@ -0,0 +1,134 @@
+import { ZoomIn, DownloadCloud, X } from "lucide-react";
+import { usePathname, useRouter } from "next/navigation";
+import {
+  VisualizationId,
+  VisualizationInformation,
+  getVisualizationInformationFromVisualizationId,
+} from "../../visualizations/visualizations";
+import { isSafeReferrerLink } from "@/utils/referrer-link-util";
+import { AppRouterInstance } from "next/dist/shared/lib/app-router-context";
+
+interface DisabledButtonConfig {
+  enabled: false;
+}
+
+type EnabledButtonConfig<AdditionalButtonInformation> = {
+  enabled: true;
+} & AdditionalButtonInformation;
+
+export type ButtonConfig<AdditionalButtonInformation> =
+  | DisabledButtonConfig
+  | EnabledButtonConfig<AdditionalButtonInformation>;
+
+export type ZoomInButtonAdditionalButtonConfig = { referrerRoute: string };
+export type DownloadButtonAdditionalButtonConfig = {};
+export type CloseButtonAdditionalButtonConfig = { referrerRoute: string | undefined | null };
+
+interface ZoomInButtonProps {
+  configuration: EnabledButtonConfig<ZoomInButtonAdditionalButtonConfig> & { id: string };
+  visualizationId: VisualizationId;
+  router: AppRouterInstance;
+}
+
+interface DownloadButtonProps {
+  configuration: EnabledButtonConfig<DownloadButtonAdditionalButtonConfig> & { id: string };
+  downloadVisualization: () => void;
+}
+
+interface CloseButtonProps {
+  configuration: EnabledButtonConfig<CloseButtonAdditionalButtonConfig> & { id: string };
+  router: AppRouterInstance;
+}
+
+interface AllButtonConfigurations {
+  zoomInButton: ButtonConfig<ZoomInButtonAdditionalButtonConfig> & { id: string };
+  downloadButton: ButtonConfig<DownloadButtonAdditionalButtonConfig> & { id: string };
+  closeButton: ButtonConfig<CloseButtonAdditionalButtonConfig> & { id: string };
+}
+
+const DownloadButton = (props: DownloadButtonProps) => (
```

Comment:
There's probably a case to be made that these buttons and their types should live in separate files. I think that would be interesting so let me know if you'd like to see that!